addons/hr_holidays/static/tests/time_off_dashboard.test.js

```diff
@@ -0,0 +1,107 @@
+import { describe, expect, test } from "@odoo/hoot";
+import { mockDate } from "@odoo/hoot-mock";
+import { defineModels, fields, getService, models, mountWebClient, onRpc } from "@web/../tests/web_test_helpers";
+import { defineHrHolidaysModels } from "./hr_holidays_test_helpers";
+
+describe.current.tags("desktop");
+
+class HrLeave extends models.Model {
+    _views = {
+        calendar: `
+            <calendar js_class="time_off_calendar_dashboard"
+                      string="Time Off Request"
+                      form_view_id="%(hr_holidays.hr_leave_view_form_dashboard_new_time_off)d"
+                      event_open_popup="1"
+                      date_start="date_from"
+                      date_stop="date_to"
+                      quick_create="0"
+                      show_unusual_days="1"
+                      color="color"
+                      hide_time="1"
+                      mode="year"
+            >
+                <field name="name"/>
+                <field name="holiday_status_id" filters="1" invisible="1" color="color"/>
+                <field name="state" invisible="1"/>
+            </calendar>
+        `,
+    };
+
+    color = fields.Integer({ related: "holiday_status_id.color" });
+    date_from = fields.Datetime();
+    date_to = fields.Datetime();
+    department_id = fields.Many2one({ relation: "hr.department" });
+    employee_id =  fields.Many2one({ relation: "hr.employee" });
+    holiday_status_id = fields.Many2one({ relation: "hr.leave.type" });
+    holiday_type = fields.Char();
+    name = fields.Char();
+    number_of_days = fields.Integer();
+    state = fields.Char();
+}
+
+class HrLeaveType extends models.Model {
+    name = fields.Char();
+    color = fields.Integer();
+}
+
+defineHrHolidaysModels();
+defineModels([HrLeave, HrLeaveType]);
+
+onRpc("hr.employee", "get_allocation_requests_amount", () => ({}));
+onRpc("hr.employee", "get_mandatory_days", () => ({}));
+onRpc("hr.employee", "get_special_days_data", () => ({ mandatoryDays: [], bankHolidays: [] }));
+onRpc("hr.leave", "get_unusual_days", () => ({}));
+onRpc("hr.leave", "has_access", () => true);
+onRpc("hr.leave.type", "get_allocation_data_request", () => []);
+onRpc("hr.leave.type", "has_accrual_allocation", () => true);
+
+test(`test employee is passed to has_accrual_allocation`, async () => {
+    onRpc("hr.leave.type", "has_accrual_allocation", ({ kwargs }) => {
+        expect.step(kwargs.context.employee_id);
+    });
+
+    await mountWebClient();
+    await getService("action").doAction({
+        id: 1,
+        res_model: "hr.leave",
+        type: "ir.actions.act_window",
+        views: [[false, "calendar"]],
+        context: { employee_id: [200] },
+        domain: [["employee_id", "in", [200]]],
+    });
+    await expect.waitForSteps([200]);
```

Comment:
can't it be expect.step?

Reply:
Mmh, yep, maybe it could be. I didn't try.